python/lib/evaluations/dmod/evaluations/specification/backend.py

```diff
@@ -102,21 +63,7 @@ class LoaderSpecification(TemplatedSpecification, abc.ABC):
     """
     Represents a class that uses a backend to load data
     """
-    __slots__ = ['_backend']
-
-    def __init__(self, backend: BackendSpecification = None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self._backend = backend
-
-    @property
-    def backend(self) -> BackendSpecification:
-        return self._backend
-
-    @abc.abstractmethod
-    def extract_fields(self) -> typing.Dict[str, typing.Any]:
-        fields = super().extract_fields()
-        fields['backend'] = self.backend.to_dict()
-        return fields
+    backend: BackendSpecification = pydantic.Field(description="Instructions on how to load data")
 
     @abc.abstractmethod
```

Comment:
Thanks for adding some docs to this. This still _feels_ wrong to me, but I think your solution is the best compromise!

Reply:
Random thought - have a secondary function that **must** be implemented that the base `__eq__` calls. Gets around the odd "Please call `super().__eq__` for the rest of the logic" situation. Something like:

```python
@abc.abstractmethod
def _model_specifics_match(self, other) -> bool:
    pass

def __eq__(self, other) -> bool:
    if not self._model_specifics_match(other):
        return False
    # Perform that other logic
```

?

Only way I can think of to get around the partial implementations of `apply_configuration` is to do some weird decorator registration scheme and that sorta seems perilous. I dunno.

Reply:
> Random thought - have a secondary function that **must** be implemented that the base `__eq__` calls.

Yeahhhh, I don't know. I think whatever makes it the most explicit and keeps the thinking local instead of global is the best way to go. I think keeping how you have it is the best way to go.